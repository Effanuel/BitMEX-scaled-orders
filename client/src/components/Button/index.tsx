import React from 'react';
// STYLES
import cx from 'classnames';
import styles from './styles.module.css';

export interface ButtonProps {
  id?: string;
  label: string;
  testID?: string;
  variant?: 'submit' | 'text' | 'custom';
  disabled?: boolean;
  onClick: (event: any) => void;
  style?: any;
  className?: any;
}

function Button({ id, label, testID, variant = 'submit', disabled, onClick, style, className = '' }: ButtonProps) {
  const buttonStyle = cx({
    [styles.button]: variant === 'submit',
    [styles.text_button]: variant === 'text',
    [className]: variant === 'custom',
  });
  return (
    <button id={id} data-test-id={testID} className={buttonStyle} style={style} disabled={disabled} onClick={onClick}>
      {label}
    </button>
  );
}

export { Button };
