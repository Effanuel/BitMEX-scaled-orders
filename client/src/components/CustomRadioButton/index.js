import React from "react";
import PropTypes from "prop-types";
import { FormCheck } from "react-bootstrap";
import styles from "./CustomRadioButton.module.css";

const CustomRadioButton = ({ label, type, name, defaultChecked }) => {
  return (
    <div className={styles.myLabel}>
      <FormCheck
        custom
        name={name}
        label={label}
        type={type}
        id={label}
        value={label}
        defaultChecked={defaultChecked}
        className={styles.noselect}
      />
    </div>
  );
};

CustomRadioButton.propTypes = {
  name: PropTypes.string,
  label: PropTypes.string,
  type: PropTypes.oneOf(["radio", "checkbox"]),
  value: PropTypes.string,
  defaultChecked: PropTypes.bool
};

export { CustomRadioButton };
