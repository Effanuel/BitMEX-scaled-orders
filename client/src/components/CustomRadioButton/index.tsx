import React from "react";
// COMPONENTS
import { FormCheck } from "react-bootstrap";
// UTILS
import styles from "./styles.module.css";

type Props = {
  label: string;
  type: "radio" | "checkbox";
  name: string;
  defaultChecked?: boolean;
};

function CustomRadioButton({ label, type, defaultChecked, name }: Props) {
  return (
    <div className={styles.label}>
      <FormCheck
        custom
        name={name}
        label={label}
        type={type}
        id={label}
        value={label}
        defaultChecked={defaultChecked}
        className={styles.label__noselect}
      />
    </div>
  );
}

export { CustomRadioButton };
