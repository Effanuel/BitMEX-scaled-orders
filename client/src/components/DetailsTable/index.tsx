import React from "react";
import { useSelector, shallowEqual } from "react-redux";

import {
  ordersSelector,
  ordersAveragePriceSelector,
  ordersRiskSelector
} from "../../redux/selectors";

import styles from "./styles.module.css";

interface IDetailsTableProps {}

export const DetailsTable: React.FunctionComponent<IDetailsTableProps> = props => {
  const { averagePrice, riskBTC } = useSelector(
    (state: any) => ({
      orders: ordersSelector(state),
      averagePrice: ordersAveragePriceSelector(state),
      riskBTC: ordersRiskSelector(state)
    }),
    shallowEqual
  );
  return (
    <table className={styles.preview}>
      <thead>
        <tr>
          <th>Details</th>
          <th>Amount</th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td>Average price:</td>
          <td>
            {`${averagePrice}`}
            <span className={styles.customStyle}> USD</span>
          </td>
        </tr>
        {riskBTC ? (
          <tr>
            <td>Risk:</td>
            <td>
              {riskBTC}
              <span className={styles.customStyle}> BTC</span>
            </td>
          </tr>
        ) : null}
      </tbody>
    </table>
  );
};
