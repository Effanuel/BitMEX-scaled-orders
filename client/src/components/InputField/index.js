import React, { memo } from 'react';
import PropTypes from 'prop-types';
import './InputField.module.css';

import { InputGroup, FormControl } from 'react-bootstrap';

const InputField = memo(({ id, label, value, onChange }) => {
  return (
    <div>
      <label htmlFor={label}>{label}</label>
      <InputGroup>
        <FormControl
          pattern="[0-9]*"
          type="number"
          id={id}
          value={value}
          onChange={onChange}
          autoComplete="off"
        />
      </InputGroup>
    </div>
  );
});

InputField.propTypes = {
  id: PropTypes.string,
  label: PropTypes.string,
  value: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  onChange: PropTypes.func
};

export { InputField };
