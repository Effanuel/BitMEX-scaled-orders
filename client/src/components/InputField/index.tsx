import * as React from 'react';
import { InputGroup, FormControl } from 'react-bootstrap';

import './InputField.module.css';

type Props<T extends object> = {
  id: string;
  label: string;
  value: any;
  onChange: any;
};

const InputField = <T extends object>(props: Props<T>) => {
  const { label, id, value, onChange } = props;
  return (
    <div>
      <label htmlFor={label}>{label}</label>
      <InputGroup>
        <FormControl
          // pattern="[0-9]*"
          type="number"
          id={id}
          value={value}
          onChange={onChange}
          autoComplete="off"
        />
      </InputGroup>
    </div>
  );
};

export { InputField };
