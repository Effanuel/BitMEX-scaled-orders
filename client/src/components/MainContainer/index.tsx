import React, {useState} from 'react';
import {Container, Grid} from '@material-ui/core';

import styles from './styles.module.css';
import {MAIN_CONTAINER} from '../../data-test-ids';

export interface Props {
  children: React.ReactNode;
  label: string;
}

export const MainContainer = React.memo(({children, label}: Props) => {
  const [maximized, setMaximized] = useState(true);

  const cornerButton = (
    <div className={styles.div_corner} key={label}>
      <div
        data-test-id={MAIN_CONTAINER.CORNER_BUTTON}
        className={styles.div_corner__button}
        onClick={() => setMaximized(!maximized)}
      />
    </div>
  );

  function maximizedView() {
    return (
      <Grid item xs container direction="row" style={{paddingBottom: '10px'}} data-test-id={MAIN_CONTAINER.MAX_VIEW}>
        <Grid item>{cornerButton}</Grid>
        <Grid item container xs style={{paddingRight: '20px', paddingTop: '5px'}}>
          <Grid container spacing={2} justify="center" alignItems="center">
            {children}
          </Grid>
        </Grid>
      </Grid>
    );
  }

  function minimizedView() {
    return (
      <div className={styles.container__row__minimized} data-test-id={MAIN_CONTAINER.MIN_VIEW}>
        {cornerButton}
        <div style={{color: 'white'}}>{label}</div>
      </div>
    );
  }

  return (
    <Container fixed maxWidth="sm" className={styles.container_scaled} style={{padding: '0px'}}>
      {maximized ? maximizedView() : minimizedView()}
    </Container>
  );
});
