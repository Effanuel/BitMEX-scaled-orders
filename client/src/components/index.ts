export * from './CustomRadioButton';
export * from './InputField';
export * from './SelectDropdown';
export * from './Spinner';
export * from './MainContainer';
export * from './PositionedSnackbar';
export * from './SVGIcon';
export * from './Button';
export {NotificationBar} from './NotificationBar';
