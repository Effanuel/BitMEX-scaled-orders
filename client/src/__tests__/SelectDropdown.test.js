import React from "react";
import { shallow } from "enzyme";
import { SelectDropdown } from "../components";

const onChange = jest.fn(),
  props = {
    instruments: ["XBTUSD", "ETHUSD", "LTCU19", "ABC_000"],
    id: "instrument",
    label: "Instrument",
    onChange
  };

const component = shallow(<SelectDropdown {...props} />);

describe("SelectDropdown component", () => {
  it("renders without crashing given the required props", () => {
    expect(component).toMatchSnapshot();
  });
  it("renders required number of buttons", () => {
    expect(component.find("option").length).toEqual(4);
  });
});
