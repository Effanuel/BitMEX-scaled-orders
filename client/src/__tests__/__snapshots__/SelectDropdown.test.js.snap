// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectDropdown component renders without crashing given the required props 1`] = `ShallowWrapper {}`;
